sparkly/session.py

```diff
@@ -89,6 +109,16 @@ def __init__(self, additional_options=None):
         options = {
             'spark.sql.catalogImplementation': 'hive',
         }
+        app_id_template = self.app_id_template
+        if app_id_template:
+            options.update({
+                'spark.app.id': app_id_template.format(
+                    process_time=self._process_time,
+                    process_uid=self._process_uid,
+                    session_time=int(time.time()),
+                    session_uid=str(uuid.uuid4()),
```

Comment:
Should we use the `hex` versions of the IDs? They're slightly shorter and do not use `-`, which might be special under some circumstances.